### DISCUSSION TITLE
Icon click Triggering map label Instead of icon

### DISCUSSION THREAD

**Discussion by it-mukaseev** [visgl/deck.gl#9061] [Q&A]
Hello,
I am rendering icons using the IconLayer in [Deck.gl](http://deck.gl/).
I'm reaching out to ask for your assistance with an issue I'm facing while integrating [Deck.gl](http://deck.gl/) v8.9.36 with Google Maps in my React project. Specifically, when a icon is placed on the map and overlaps with a map label, clicking on the icon does not function as expected. Instead of selecting the icon, the map's info window for the underlying label opens
I would greatly appreciate any guidance or suggestions you could provide to help resolve this issue.

https://github.com/user-attachments/assets/1c0af7ca-86c5-4193-8b8e-025aeb48c80f

**Comment by Pessimistress** [comment]
The Deck overlay receives the click event by registering a map event handler: `map.addListener('click', cb)` It has no visibility into how the other parts of the map respond to the event.

You can access the original Google Maps event object with:

```js
onClick: (pickInfo, event) => {
  console.log(event.srcEvent); // Google maps click event
}
```

**Comment by boltex** [comment]
@it-mukaseev Did you ever found a solution?
